core/modules/qdisp/QueryResource.cc

```diff
@@ -56,8 +56,12 @@ void QueryResource::ProvisionDone(XrdSsiSession* s) { // Step 3
     LOGF_INFO("Provision done");
     if(!s) {
         // Check eInfo in resource for error details
-        int code;
-        char const* msg = eInfo.Get(code);
+        int code = -1;
+        char const* message = eInfo.Get(code);
+        std::string msg = "no message from XrdSsi, code may not be reliable";
```

Comment:
eInfo.Get() is supposed to return and error message and set code to the error id from xrootd. If the returned value is NULL, xrootd (XrdSsi) did not supply an error message, and there's reason to believe 'code' is invalid. For unit testing, it is no surprise that xrootd doesn't have an error code available as we haven't actually asked it to do anything.